agroal-pool/src/main/java/io/agroal/pool/ConnectionPool.java

```diff
@@ -207,7 +219,7 @@ public void close() {
         allConnections.clear();
         activeCount.reset();
 
-        synchronizer.release( synchronizer.getQueueLength() );
+        while ( handlerTransferQueue.tryTransfer( TRANSFER_POISON ) ) ; // Unblock waiting threads with CancellationException
```

Comment:
this always expect some thread to be parked waiting for it; is it correct?
If there are no waiting thread this should just leave this, without looping?